src/driver/mysql/MysqlQueryRunner.ts

```diff
@@ -949,28 +957,54 @@ export class MysqlQueryRunner extends BaseQueryRunner implements QueryRunner {
      * Creates a new check constraint.
      */
     async createCheckConstraint(tableOrName: Table|string, checkConstraint: TableCheck): Promise<void> {
-        throw new Error(`MySql does not support check constraints.`);
+        const table = tableOrName instanceof Table ? tableOrName : await this.getCachedTable(tableOrName);
+
+        // new unique constraint may be passed without name. In this case we generate unique name manually.
+        if (!checkConstraint.name)
+            checkConstraint.name = this.connection.namingStrategy.checkConstraintName(table.name, checkConstraint.expression!);
+
+        const up = this.createCheckConstraintSql(table, checkConstraint);
+        const down = this.dropCheckConstraintSql(table, checkConstraint);
+        await this.executeQueries(up, down);
+        table.addCheckConstraint(checkConstraint);
     }
 
     /**
      * Creates a new check constraints.
      */
     async createCheckConstraints(tableOrName: Table|string, checkConstraints: TableCheck[]): Promise<void> {
-        throw new Error(`MySql does not support check constraints.`);
+        if (await this.getVersion() < "8.0.16") {
+            throw new Error(`MySql check constraints are supported since version 8.0.16`);
+        }
+
+        const promises = checkConstraints.map(checkConstraint => this.createCheckConstraint(tableOrName, checkConstraint));
+        await Promise.all(promises);
     }
 
     /**
      * Drops check constraint.
      */
     async dropCheckConstraint(tableOrName: Table|string, checkOrName: TableCheck|string): Promise<void> {
-        throw new Error(`MySql does not support check constraints.`);
+        const table = tableOrName instanceof Table ? tableOrName : await this.getCachedTable(tableOrName);
```

Comment:
Needs version check

Reply:
Fixed the comments, thank you very much. And also tests written by me passed. But unfortunately some random tests fail and to be honest I didn't find how to fix them. Can you please give me a hit on how to fix them?

Reply:
@imnotjames bumping vmuresanu's request!
This feature addition would be greatly appreciated :)